app/Http/Controllers/V2/Nurseries/CreateNurseryWithFormController.php

```diff
@@ -31,17 +29,16 @@ public function __invoke(CreateEntityFormRequest $formRequest)
             'status' => EntityStatusStateMachine::STARTED,
         ]);
 
-        $lastTask = Task::where('project_id', $project->id)
-            ->orderBy('created_at', 'desc')
-            ->first();
+        $lastTask = $project->tasks()->orderby('due_at', 'desc')->first();
 
         if ($lastTask) {
-            $nextReportDueDate = Carbon::parse($lastTask->due_at)->addWeeks(4);
+            $nextReportingPeriod = Carbon::parse($lastTask->due_at)->addWeeks(4);
```

Comment:
I think this `addWeeks` is incorrect. With the `$weeksDifference` below, this is effectively checking the `due_at` date against `Carbon::now()`

I think a better way to arrange this comparison is:
```
if (Carbon::now() <= $lastTask->due_at->subWeeks(4)) {
```